### PR TITLE
new package: hedgewars

### PR DESCRIPTION
**Pull request by skrzyp**
Finally :)
One of my favourite games, closes one of many tasks in #2097

### CODE REVIEW

File: srcpkgs/hedgewars/template [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Template file for 'hedgewars'
+pkgname=hedgewars
+version=0.9.21.1
+revision=1
+_version=0.9.21
+wrksrc="${pkgname}-src-${_version}"
+build_style=cmake
+configure_args="
+	-DCMAKE_BUILD_TYPE=\"RELEASE\"
```
Comment (xtraeme) [inline srcpkgs/hedgewars/template]: This is set by default with `build_style=cmake`

File: srcpkgs/hedgewars/template [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	-DNOSERVER=1
+	-DCMAKE_INSTALL_PREFIX=/usr
```
Comment (xtraeme) [inline srcpkgs/hedgewars/template]: This is set by default with `build_style=cmake`

File: srcpkgs/hedgewars/template [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	-DDATA_INSTALL_DIR=/usr/share/${pkgname}
+	-DCMAKE_VERBOSE_MAKEFILE=TRUE
+	-DPHYSFS_SYSTEM=on
+"
+hostmakedepends="cmake lua lua51 pkg-config"
+makedepends="fpc qt-qmake qt-devel SDL-devel SDL_net-devel SDL_mixer-devel SDL_image-devel SDL_ttf-devel lua51-devel physfs-devel libpng-devel ffmpeg-devel libfreeglut-devel"
```
Comment (xtraeme) [inline srcpkgs/hedgewars/template]: `fpc` must be really in `hostmakedepends` instead.

Comment (xtraeme) [inline srcpkgs/hedgewars/template]: `qt-qmake` too.

File: srcpkgs/hedgewars/template [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# Template file for 'hedgewars'
+pkgname=hedgewars
+version=0.9.21.1
+revision=1
+_version=0.9.21
```
Comment (xtraeme) [inline srcpkgs/hedgewars/template]: Remove this var (`_version`) and use `${version%.*}` instead where it's necessary.

File: common/shlibs [hunk 5]
```diff
@@ -2163,3 +2163,5 @@ libgflags.so.2 gflags-2.1.2_1
 libgflags_nothreads.so.2 gflags-2.1.2_1
 libswmhack.so.0.0 spectrwm-2.7.2_1
 libchardet.so.1 libchardet-1.0.4_1
+libphyslayer.so hedgewars-0.9.21_1
```
Comment (xtraeme) [inline common/shlibs]: These shared libraries are not meant to be used as linkable... therefore there's no need to add them.

File: srcpkgs/hedgewars/template [hunk 6]
```diff
@@ -0,0 +1,23 @@
+# Template file for 'hedgewars'
+pkgname=hedgewars
+version=0.9.21.1
+revision=1
+wrksrc="${pkgname}-src-${version%.*}"
+build_style=cmake
+configure_args="
+	-DCMAKE_BUILD_TYPE=\"RELEASE\"
```
Comment (xtraeme) [inline srcpkgs/hedgewars/template]: `CMAKE_BUILD_TYPE=release` is set already by `build_style=cmake`

See `common/build-style/cmake.sh`

**Comment by xtraeme** [comment]
Please make a `hedgewars-data` subpkg that contains the `data` files (`/usr/share/hedgewars/Data`) and it's set as `noarch`.

File: srcpkgs/hedgewars/template [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+license="GPL-2"
+homepage="http://hedgewars.org"
+distfiles="http://download.gna.org/hedgewars/hedgewars-src-${version}.tar.bz2"
+checksum=590a9458d2123c5550d5eb39edfed28d0663703d76e13c8987237c92ca41f3f8
+
+hedgewars-data_package() {
+	short_desc+=" - data files"
+	noarch=yes
+	do_install() {
+		vmove usr/share/hedgewars
```
Comment (xtraeme) [inline srcpkgs/hedgewars/template]: - `vmove usr/share/hedgewars/Data` should be enough.
- The .desktop file must be owned by the main pkg, not `-data`.

File: srcpkgs/hedgewars/template [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+post_install() {
+	vinstall ${DESTDIR}/usr/share/hedgewars/Data/misc/hedgewars.desktop 644 usr/share/applications/hedgewars.desktop
+}
+
+hedgewars-data_package() {
+	short_desc+=" - data files"
+	noarch=yes
+	do_install() {
+		vmove usr/share/hedgewars/data
```
Comment (xtraeme) [inline srcpkgs/hedgewars/template]: Should be `Data` not `data`, otherwise it's empty.

File: srcpkgs/hedgewars/template [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+depends="dejavu-fonts-ttf wqy-microhei hedgewars-data"
+short_desc="Funny turn-based artillery game, featuring fighting Hedgehogs!"
+maintainer="Jakub Skrzypnik <jot.skrzyp@gmail.com>"
+license="GPL-2"
+homepage="http://hedgewars.org"
+distfiles="http://download.gna.org/hedgewars/hedgewars-src-${version}.tar.bz2"
+checksum=590a9458d2123c5550d5eb39edfed28d0663703d76e13c8987237c92ca41f3f8
+
+post_install() {
+	vinstall ${DESTDIR}/usr/share/hedgewars/Data/misc/hedgewars.desktop 644 usr/share/applications/hedgewars.desktop
```
Comment (xtraeme) [inline srcpkgs/hedgewars/template]: This is wrong, should be:

```
vinstall ${DESTDIR}/usr/share/hedgewars/Data/misc/hedgewars.desktop 644 usr/share/applications
```

Comment (Jaybri) [inline srcpkgs/hedgewars/template]: Sent a reply,from my smartphone. Will get message right away, so the reply will be soon. Thanks for your patience. :0)                                                                                                                                                                                                                  From: Juan RPSent: Saturday, September 19, 2015 2:03 AMTo: voidlinux/void-packagesReply To: voidlinux/void-packagesSubject: Re: [void-packages] new package: hedgewars (#2532)In srcpkgs/hedgewars/template:
> +	-DDATA_INSTALL_DIR=/usr/share/${pkgname}
> +	-DCMAKE_VERBOSE_MAKEFILE=TRUE
> +	-DPHYSFS_SYSTEM=on
> +"
> +hostmakedepends="cmake lua lua51 pkg-config fpc qt-qmake"
> +makedepends="qt-devel SDL-devel SDL_net-devel SDL_mixer-devel SDL_image-devel SDL_ttf-devel lua51-devel physfs-devel libpng-devel ffmpeg-devel libfreeglut-devel"
> +depends="dejavu-fonts-ttf wqy-microhei hedgewars-data"
> +short_desc="Funny turn-based artillery game, featuring fighting Hedgehogs!"
> +maintainer="Jakub Skrzypnik <jot.skrzyp@gmail.com>"
> +license="GPL-2"
> +homepage="http://hedgewars.org"
> +distfiles="http://download.gna.org/hedgewars/hedgewars-src-${version}.tar.bz2"
> +checksum=590a9458d2123c5550d5eb39edfed28d0663703d76e13c8987237c92ca41f3f8
> +
> +post_install() {
> +	vinstall ${DESTDIR}/usr/share/hedgewars/Data/misc/hedgewars.desktop 644 usr/share/applications/hedgewars.desktop

This is wrong, should be:

vinstall ${DESTDIR}/usr/share/hedgewars/Data/misc/hedgewars.desktop 644 usr/share/applications


—Reply to this email directly or view it on GitHub.

**Comment by skrzyp** [comment]
It looks like @xtraeme actually doesn't like me…

File: srcpkgs/hedgewars/template [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+checksum=590a9458d2123c5550d5eb39edfed28d0663703d76e13c8987237c92ca41f3f8
+
+post_install() {
+	vinstall ${DESTDIR}/usr/share/hedgewars/Data/misc/hedgewars.desktop 644 usr/share/applications
+}
+
+hedgewars-data_package() {
+	short_desc+=" - data files"
+	noarch=yes
+	do_install() {
```
Comment (xtraeme) [inline srcpkgs/hedgewars/template]: `do_install` -> `pkg_install`

**Comment by Gottox** [comment]
Fix the errors, get pulled upstream

**Pull request merged**